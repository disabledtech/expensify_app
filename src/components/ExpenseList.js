import React from "react";
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import SelectExpenses from '../selectors/expenses';

//Named export for tests
export const ExpenseList = (props) => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-big-screen">Expense</div>
            <div className="show-for-big-screen">Amount</div>
        </div>
        <div className="list-body">
            {
                props.expenses.length === 0 ? (
                    <div>
                        <span className="list-item list-item--message">
                            No expenses.
                        </span>
                    </div>
                ) : (
                    props.expenses.map ((expense) => (
                        <ExpenseListItem
                            key={expense.id}
                            {...expense}
                        />
                ) 
            ))
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: SelectExpenses(state.expenses, state.filters )
    }
}

export default connect(mapStateToProps)(ExpenseList);
