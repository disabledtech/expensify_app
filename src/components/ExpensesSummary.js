import React from "react";
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
    const pluralExpenseOrNot = expenseCount === 1 ? 'expense': 'expenses';
    const total = numeral(expenseTotal / 100).format('$0,0.00');

    return (
        <div>
            {   
                <h2>Viewing a total of { expenseCount } { pluralExpenseOrNot } for { total }.</h2>   
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
