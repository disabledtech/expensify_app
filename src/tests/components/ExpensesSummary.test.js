import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixture/expenses';

test('Should render ExpenseSummary for 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={9271} />);
    expect(wrapper).toMatchSnapshot();   
});

test('Should render ExpenseSummary for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expenseTotal={9222171} />);
    expect(wrapper).toMatchSnapshot();   
});
