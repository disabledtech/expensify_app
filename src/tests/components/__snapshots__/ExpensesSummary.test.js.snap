// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseSummary for 1 expense 1`] = `
<div>
  <h2>
    Viewing a total of 
    1
     
    expense
     for 
    $92.71
    .
  </h2>
</div>
`;

exports[`Should render ExpenseSummary for multiple expenses 1`] = `
<div>
  <h2>
    Viewing a total of 
    3
     
    expenses
     for 
    $92,221.71
    .
  </h2>
</div>
`;
